plugin/trino-oracle/src/test/java/io/trino/plugin/oracle/AbstractTestOracleTypeMapping.java

```diff
@@ -733,6 +734,40 @@ public void testTimestamp(ZoneId sessionZone)
                 .execute(getQueryRunner(), session, trinoCreateAndInsert(session, "test_timestamp"));
     }
 
+    @Test(dataProvider = "sessionZonesDataProvider")
+    public void testNanoTimestamp(ZoneId sessionZone)
+    {
+        Session session = Session.builder(getSession())
+                .setTimeZoneKey(getTimeZoneKey(sessionZone.getId()))
+                .build();
+
+        SqlDataTypeTest.create()
+                // min value in Oracle
+                .addRoundTrip("TIMESTAMP(9)", "TIMESTAMP '-4712-01-01 00:00:00.000000000'", TIMESTAMP_NANOS, "TIMESTAMP '-4712-01-01 00:00:00.000000000'")
```

Comment:
Looks like we are not able to insert something in oracle beyond 9 precision, because 9 is max precision.
During insert we got exception: ORA-01861: literal does not match format string

Reply:
Not sure how you tested, but I guess you simplify add the new roundtip here. It doesn't work because these roundtip run on `oracleCreateAndInsert`. Please separate a test method and use only `trinoCreateAsSelect` and `trinoCreateAndInsert`.

Reply:
For `trinoCreateAsSelect` and `oracleCreateAndInsert` we have:
ORA-00972: identifier is too long

Reply:
Test looks like this:

```
        SqlDataTypeTest.create()
                .addRoundTrip("TIMESTAMP(9)", "TIMESTAMP '2019-03-18 10:01:17.98798798711'", TIMESTAMP_NANOS, "TIMESTAMP '2019-03-18 10:01:17.987987987'")
                .execute(getQueryRunner(), trinoCreateAsSelect("test_timestamp_nano_beyond"))
                .execute(getQueryRunner(), trinoCreateAndInsert("test_timestamp_nano_beyond"));
```

Reply:
`testTimestampNanosBeyondPrecision` added